### PR TITLE
Rotate Animation for Interactable and Progress Manager

### PR DESCRIPTION
**Pull request by chadleytan**
Main Changes:
* Added a rotate animation coroutine `RotateAnimation(GameObject obj, Quaternion desiredRotation, float turnSpeed)`
  * It is used for animating player rotation when interacting with an interactable
  * It can also be used to animate the rotation of an interactable (NPC) on interaction
* Made the animation of the interactable text a bit more smooth
* Started implementing `ProgressManager.cs` and was able to make it work with a stamp collectibles
  * This keeps track of unlocked status of all stamps through a dictionary
  * Communicate with the dialogue manager on what dialogue to show based on progression (TODO)
  * `Collectible.cs` is a parent class for any collectible which just has a variable of collectible name. Made child class `StampCollectible.cs` which has a sprite variable that you drag into the inspector (I'll make sure to include in Wiki)

### CODE REVIEW

File: Assets/Scripts/ProgressManager.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[System.Serializable]
```
Comment (jurcaua) [inline Assets/Scripts/ProgressManager.cs]: Why do we need to make this serializable?

File: Assets/Scripts/ProgressManager.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+
+[System.Serializable]
+public class ProgressManager: Singleton<ProgressManager>
+{
+    // Comments below is done in Mission Manager
+        // Keep track of all obtained missions
+        // Mission is obtained from an interactable
+        // Missions should also be an interactable which checks if mission is obtained (must talk with group)
```
Comment (jurcaua) [inline Assets/Scripts/ProgressManager.cs]: Oh, why might missions need to be an interactable?

**Review by jurcaua — COMMENTED** [review]
Some beautiful code right here, the ProgressManager is looking real nice, and I'm liking the support functions for possible future use. Great comments too.

Small things I'm curious about for which I left some comments, but otherwise I have no complaints! Great stuff!

File: Assets/Scripts/ProgressManager.cs (hunk 1 above)
Comment (chadleytan) [reply Assets/Scripts/ProgressManager.cs]: Oops forgot to take that out

File: Assets/Scripts/ProgressManager.cs (hunk 2 above)
Comment (chadleytan) [reply Assets/Scripts/ProgressManager.cs]: It was the whole thing we were talking about if triggering the mission needed to be an interactable or just something on enter and indicated by the UI. Left that comment there for more of a reminder

Comment (jurcaua) [reply Assets/Scripts/ProgressManager.cs]: Still not 100% convinced about it, cause the mission itself I imagine probably should not be an interactable, but an interactable (such as the end of an NPC interaction) will start the mission.

**Review by jurcaua — APPROVED** [review]

File: Assets/Scripts/ProgressManager.cs (hunk 2 above)
Comment (chadleytan) [reply Assets/Scripts/ProgressManager.cs]: Sorry, it was a poorly made comment but to clarify, I was referring to the entrance of the mission that should be the interactable and not the mission itself. I'll remove these comments to reduce confusion.

**Pull request merged**